pkg/serviceaccounts/controllers/docker_registry_service.go

```diff
@@ -242,16 +285,37 @@ func (e *DockerRegistryServiceController) getDockerRegistryLocations() []string
 		return []string{}
 	}
 	service := obj.(*kapi.Service)
+	result := []string{}
 
 	hasClusterIP := (len(service.Spec.ClusterIP) > 0) && (net.ParseIP(service.Spec.ClusterIP) != nil)
 	if hasClusterIP && len(service.Spec.Ports) > 0 {
-		return []string{
+		result = append(result, []string{
 			net.JoinHostPort(service.Spec.ClusterIP, fmt.Sprintf("%d", service.Spec.Ports[0].Port)),
 			net.JoinHostPort(fmt.Sprintf("%s.%s.svc", service.Name, service.Namespace), fmt.Sprintf("%d", service.Spec.Ports[0].Port)),
+		}...)
+	}
+
+	// Generate dockercfg entry for value of OPENSHIFT_DEFAULT_REGISTRY (if set)
```

Comment:
In most cases users will want to expose registry via route and set `OPENSHIFT_DEFAULT_REGISTRY` to that hostname so builds will use that registry for pushing. That should work well.

Users that don't want route, but still use named internal service can set this to `docker-registry.default.svc.cluster.local:5000` and it should work for them as well.